### PR TITLE
Geopy 1013

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1013 - Manage warnings on geoimage copy_from_extent**
Managing warnings.
Trying to supress the internal warnings.
Catch the warnings in the test.

### CODE REVIEW

File: geoh5py/data/primitive_type_enum.py [hunk 1]
```diff
@@ -39,3 +43,23 @@ class PrimitiveTypeEnum(Enum):
     GEOMETRIC = 9
     MULTI_TEXT = 10
     BOOLEAN = 11
+
+
+def convert_to_primitive_type(data: np.ndarray, primitive_type: str) -> np.ndarray:
```
Comment (domfournier) [inline geoh5py/data/primitive_type_enum.py:48]: I like the idea a lot. See comment above so we can leverage this in the values setter of NumericData.

File: geoh5py/objects/drillhole.py [hunk 2]
```diff
@@ -419,6 +420,11 @@ def add_data(
                     continue
                 kwargs[key] = val
 
+            if "values" in kwargs:
+                kwargs["values"] = convert_to_primitive_type(
+                    kwargs["values"], entity_type["primitive_type"]
```
Comment (domfournier) [inline geoh5py/objects/drillhole.py:425]: Maybe we could let the setter do this?

**Review by domfournier — COMMENTED** [review]
I like the new functionality. Let's see if we can simplify the values setter

File: geoh5py/data/numeric_data.py [hunk 3]
```diff
@@ -89,8 +89,8 @@ def check_vector_length(self, values: np.ndarray | None) -> np.ndarray:
                 warn("Input 'values' converted to a 1D array.")
 
             if len(values) < self.n_values:
-                full_vector = np.ones(self.n_values, dtype=type(self.ndv))
-                full_vector *= np.nan if isinstance(self.ndv, float) else self.ndv
+                full_vector = np.ones(self.n_values, dtype=values.dtype)
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/data/numeric_data.py]: major changes in numerci data classes

File: geoh5py/data/primitive_type_enum.py (hunk 1 above)
Comment (MatthieuCMira) [reply geoh5py/data/primitive_type_enum.py]: I changed convert_to_primitive_type in the numeric Datatype.

In order to respect the type of the data. 


New "check_type" function. 
Float can take any number,
Integer can take number without decimal.
Booleans can just take 0,1 (or TRUE, FALSE).

function ckeck_vector_length change to "format_value" in order to do everything.

**Comment by MatthieuCMira** [comment]
Why data is formatted only if "self.association" not in data_association in data.py?

File: geoh5py/data/numeric_data.py [hunk 4]
```diff
@@ -40,57 +39,48 @@ def primitive_type(cls) -> PrimitiveTypeEnum:
...
+        Check if the type of values is valid and convert it to right dtype.
+        :param values: numpy array to modify.
+        :return: the formatted values.
+        """
+
     @property
     def values(self) -> np.ndarray | None:
         """
-        :return: values: An array of float values
+        :return: values: An array of integer values
```
Comment (domfournier) [inline geoh5py/data/numeric_data.py:53]:
```suggestion
        :return: values: An array of values
```

File: geoh5py/data/integer_data.py [hunk 5]
```diff
@@ -27,48 +25,31 @@
 
 
 class IntegerData(NumericData):
+    def format_type(self, values: np.ndarray) -> np.ndarray:
+        """
+        Check if the type of values is valid and convert it to right dtype.
```
Comment (domfournier) [inline geoh5py/data/integer_data.py:30]:
```suggestion
        Check if the type of values is valid and coerse to type int32.
```

File: geoh5py/data/float_data.py [hunk 6]
```diff
@@ -27,10 +29,28 @@ class FloatData(NumericData):
     Data container for floats values
     """
 
+    def format_type(self, values: np.ndarray) -> np.ndarray:
+        """
+        Check if the type of values is valid and convert it to right dtype.
```
Comment (domfournier) [inline geoh5py/data/float_data.py:34]:
```suggestion
        Check if the type of values is valid and coerse to type float64.
```

File: geoh5py/data/boolean_data.py [hunk 7]
```diff
@@ -35,46 +35,26 @@ def __init__(self, data_type: DataType, **kwargs):
...
-    @classmethod
-    def primitive_type(cls) -> PrimitiveTypeEnum:
-        return PrimitiveTypeEnum.BOOLEAN
-
-    @property
-    def values(self) -> np.ndarray | None:
+    def format_type(self, values: np.ndarray):
         """
-        :return: values: An array of integer values
+        Check if the type of values is valid and convert it to right dtype.
```
Comment (domfournier) [inline geoh5py/data/boolean_data.py:40]:
```suggestion
        Check if the type of values is valid and coerse to type bool.
```

File: geoh5py/data/numeric_data.py [hunk 8]
```diff
@@ -40,57 +39,48 @@ def primitive_type(cls) -> PrimitiveTypeEnum:
...
                 f"Input 'values' for {self} must be of type {np.ndarray} or None."
             )
 
-        self._values = values
-
+        self._values = self.format_values(values)
         self.workspace.update_attribute(self, "values")
 
-    def check_vector_length(self, values: np.ndarray | None) -> np.ndarray:
+    def format_lenght(self, values: np.ndarray) -> np.ndarray:
```
Comment (domfournier) [inline geoh5py/data/numeric_data.py:73]:
```suggestion
    def format_length(self, values: np.ndarray) -> np.ndarray:
```

File: geoh5py/data/numeric_data.py [hunk 9]
```diff
@@ -99,4 +89,36 @@ def check_vector_length(self, values: np.ndarray | None) -> np.ndarray:
...
+
+        if values.ndim > 1:
+            values = np.ravel(values)
+            warn("Input 'values' converted to a 1D array.")
+
+        # change nan values to nan_value
+        values[np.isnan(values)] = self.nan_value
+
+        # check the length of the values
+        values = self.format_lenght(values)
```
Comment (domfournier) [inline geoh5py/data/numeric_data.py:119]:
```suggestion
        values = self.format_length(values)
```

**Review by domfournier — COMMENTED** [review]
Super, great job on the refactoring. See comments.

**Review by domfournier — APPROVED** [review]
Looking good, going to test

**Pull request merged**